### DISCUSSION TITLE
Using PicoDebug programmer on Apple Silicon - workaround

### DISCUSSION THREAD

**Discussion by WayneKeenan** [earlephilhower/arduino-pico#1743] [Show and tell]
Hi,

I thought this might help other Apple Silicon users.

The bundled `openocd` is an Intel binary and trying to get libusb & libhid installed in an `arch x86_64` `homebrew` environment is just a disaster...

To get this to work here's what I did, whilst the IDE is not running:

1. Install Apple silicon hombrew & `openocd`   (I used my already existing setup so don't have detail 'replay instructions')

2. Edit:		`rp2040/hardware/rp2040/3.6.0/boards.txt`

Comment out this line, the flags cause linking with `Serial` to fail.

```
#rpipicow.menu.uploadmethod.picodebug.build.picodebugflags=-UUSE_TINYUSB -DNO_USB -DDISABLE_USB_SERIAL -I{runtime.platform.path}/tools/libpico
```

Add this below:

```
rpipicow.menu.uploadmethod.picodebug.build.picodebugflags=-DENABLE_PICOTOOL_USB
```

3. Edit:  `rp2040/hardware/rp2040/3.6.0/platform.txt`

Point the Arduino programming config to the Apple silicon Homebrew and custom `openocd` config file (adjust the `-f` path accordingly):

Comment out this line:
```
tools.picodebug.upload.pattern="{cmd}/bin/openocd" -f "board/pico-debug.cfg" -s "{cmd}/share/openocd/scripts" -c "program {{build.path}/{build.project_name}.elf} verify reset exit"
```

Add this below:
```
tools.picodebug.upload.pattern="/opt/homebrew/bin/openocd" -s /opt/homebrew/share/openocd/scripts -f /Users/wayne/projects/pico/cfg/openocd_rp2040.cfg -c "tcl_port disabled" -c "gdb_port disabled" -c "tcl_port disabled"  -c "program {{build.path}/{build.project_name}.elf}" -c reset -c shutdown
```

4. Unzip and put the attached `openocd`  somewhere safe and make sure the path in step 3 points to it.

The `openocd_rp2040.cfg` comes from a pico-sdk & pico debug setup about 6 months ago, it might be modified, I can't recall, but this works for me here in Arduino and in Clion.

Hope it helps.

[openocd_rp2040.cfg.zip](https://github.com/earlephilhower/arduino-pico/files/12777057/openocd_rp2040.cfg.zip)

**Comment by earlephilhower** [comment]
Thanks for the detailed steps.  I'll pin this as a discussion so it'll be easier to find.

**Comment by ruswerner** [comment]
@WayneKeenan Would you mind sharing your CLion setup for using arduino-pico? I've hacked together some shoddy cmake files that enable code completion, but I've just added some custom targets that shell out to `arduino-cli` to compile and upload. I'd also like to get openocd debugging working, but haven't gotten that far yet. Any help, even if it's a zip of a project folder that has all the config in it, would be extremely helpful. I'm on a M1 mac using CLion, targetting pico & pico w. Cheers mate.

**Reply by WayneKeenan** [reply]
@ruswerner  I'm using the Pico SDK directly in CLion.
